modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/utils/APIExportUtil.java

```diff
@@ -812,4 +821,73 @@ private static void writeFile(String path, String content) throws APIExportExcep
 
     }
 
+    /**
+     * Checks whether the request is violating cross tenant permission policies. Cross tenant resource access is allowed
+     * only for the super tenant admin user, only if the server is started with 'migrationMode=true' system property set.
+     *
+     * @param apiDomain Tenant domain of the API's provider
+     * @param username  Logged in user name
+     * @return Whether cross tenant access policies violated
+     * @throws APIExportException
+     */
+    public static Boolean isCrossTenantAccessPermissionsViolated(String apiDomain, String username) throws APIExportException {
+        String resourceRquesterDomain = MultitenantUtils.getTenantDomain(username);
+        if (resourceRquesterDomain.equals(apiDomain)) {
+            return false;
+        }
+        String superAdminRole;
+        try {
+            superAdminRole = ServiceReferenceHolder.getInstance().getRealmService().
+                    getTenantUserRealm(MultitenantConstants.SUPER_TENANT_ID).getRealmConfiguration().getAdminRoleName();
+        } catch (UserStoreException e) {
+            String errorMsg = "Error in getting super admin role name";
+            throw new APIExportException(errorMsg, e);
```

Comment:
Don't we need to log the exception here?

Reply:
Logging and then throwing will cause logging the same error message multiple times. Error is 'catched' here to pass information about the particular error occurred in the try block, to the upper level, where it is logged.